web/src/invite.ts

```diff
@@ -347,8 +350,22 @@ function open_invite_user_modal(e: JQuery.ClickEvent<Document, undefined>): void
 
         const user_has_email_set = !settings_data.user_email_not_configured();
 
-        set_custom_time_inputs_visibility();
-        set_expires_on_text();
+        settings_components.set_custom_time_inputs_visibility(
+            $expires_in,
+            custom_expiration_time_unit,
+            custom_expiration_time_input,
+        );
+        let expiration_time_in_minutes;
+        if ($expires_in.val() !== "custom") {
+            expiration_time_in_minutes = Number.parseFloat($expires_in.val()!);
+        } else {
+            expiration_time_in_minutes = util.get_custom_time_in_minutes(
```

Comment:
So, per my comment above, I think this could all happen in `valid_to` directly. And you'd just pass `$expires_in.val()!` to that function.